tests/network/vmi_istio.go

```diff
@@ -0,0 +1,297 @@
+	istioInjectNamespaceLabel    = "istio-injection"
+	vmiAppSelector               = "istio-vmi-app"
+	svcDeclaredTestPort          = 1500
+	svcUndeclaredTestPort        = 1501
+)
+
+var (
+	vmi     *v1.VirtualMachineInstance
+	service *corev1.Service
+)
+
+func istioServiceMeshDeployed() bool {
+	value := os.Getenv(istioDeployedEnvVariable)
+	if value == "true" {
+		return true
+	}
+	return false
+}
+
+func withAppLabel(value string) libvmi.Option {
+	return func(vmi *v1.VirtualMachineInstance) {
+		if vmi.Labels == nil {
+			vmi.Labels = map[string]string{}
+		}
+		vmi.Labels["app"] = value
+	}
+}
+
+func withIstioAnnotation() libvmi.Option {
+	return func(vmi *v1.VirtualMachineInstance) {
+		if vmi.Annotations == nil {
+			vmi.Annotations = map[string]string{}
+		}
+		vmi.Annotations[istioInjectSidecarAnnotation] = "true"
+	}
+}
+
+func newServiceWithAppSelector(name, namespace, selector string) *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: map[string]string{
+				"app": selector,
+			},
+			Ports: []corev1.ServicePort{
+				{
+					Port: svcDeclaredTestPort,
+				},
+			},
+		},
+	}
+}
+
+func newRandomCirrosVMIWithIstioSidecarAndDefaultServiceAccountAndNetworkAndInterface(appLabel string, network *v1.Network, ports []v1.Port) *v1.VirtualMachineInstance {
+	vmi := libvmi.NewCirros(
+		libvmi.WithNetwork(network),
+		libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding(ports...)),
+		withIstioAnnotation(),
+		withAppLabel(appLabel),
+	)
+	// Istio-proxy requires service account token to be mounted
+	tests.AddServiceAccountDisk(vmi, "default")
+	return vmi
+}
+
+var _ = SIGDescribe("[Serial] Istio", func() {
+	BeforeEach(func() {
+		if !istioServiceMeshDeployed() {
+			Skip("Istio service mesh is required for service-mesh tests to run")
+		}
+	})
+	var (
+		err        error
+		virtClient kubecli.KubevirtClient
+	)
+	network := &v1.Network{
+		Name: "default",
+		NetworkSource: v1.NetworkSource{
+			Pod: &v1.PodNetwork{},
+		},
+	}
+
+	Context("Virtual Machine with masquerade interface and explicitly specified ports", func() {
+		verifyClientServerConnectivity := func(serverVMI *v1.VirtualMachineInstance, targetPort int) (*batchv1.Job, error) {
+			By("Starting HTTP Server")
+			tests.StartHTTPServer(vmi, targetPort)
+
+			By("Getting back the VMI IP")
+			vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			vmiIP := vmi.Status.Interfaces[0].IP
+
+			By("Running job to send a request to the server")
+			return virtClient.BatchV1().Jobs(tests.NamespaceTestDefault).Create(
+				context.Background(),
+				generateBatchJobSpec(vmiIP, fmt.Sprintf("%d", targetPort), tests.NewHelloWorldJobHTTP),
+				metav1.CreateOptions{},
+			)
+		}
+		BeforeEach(func() {
+			tests.BeforeTestCleanup()
```

Comment:
This doesn't remove the namespaces, it only cleans bunch of resource types. So it should be pretty much no-op.
It shouldn't ever clean anything created by me, unless I'm missing something, so it's here only to clean potential leftovers.
I put it here because I saw this used in `vmi_networking.go` but as far as I'm concerned, this can be removed.

Reply:
It doesn't remove the namespaces, but it does delete pretty much everything left in them: https://github.com/kubevirt/kubevirt/blob/1ed6b509e32577c253c3cca2e83d04ff008bf965/tests/utils.go#L1650

This is nuking the playground after you're done with it. I'm just asking if this was what we've decided long ago. I think so, but I also remember an effort of cleaning up each entity - I don't remember the reasons why we've stopped doing that.

Reply:
Removed the `tests.BeforeTestCleanup()` and added `AfterEach` for Inbound tests to delete the Job.

Reply:
I am unsure this is what we want to do... That was why I asked @AlonaKaplan what her view on this is, because I remember we've spoken about such things in the past, and have reached some sort of consensus. 

You might have to put this as it was, but for now, let's wait for her answer.